outbox.js

```diff
@@ -0,0 +1,205 @@
+
+    this.renderedItems.clear();
+  }
+}
+
+// Register the new element with the browser
+customElements.define("distributed-outbox", DistributedOutbox);
+
+class DistributedActivity extends HTMLElement {
+  constructor() {
+    super();
+    this.activityType = "";
+    this.activityData = {};
+  }
+
+  static get observedAttributes() {
+    return ["type", "data"];
+  }
+
+  connectedCallback() {
+    this.activityType = this.getAttribute("type");
+    this.activityData = JSON.parse(this.getAttribute("data"));
+    this.renderActivity();
+  }
+
+  async fetchAndDisplayPost(postUrl) {
+    try {
+      const response = await fetch(postUrl);
+      if (!response.ok) {
+        throw new Error(`HTTP error! Status: ${response.status}`);
+      }
+      const postData = await response.json();
+      this.displayPostContent(postData.content);
+    } catch (error) {
+      console.error("Error fetching post content:", error);
+    }
+  }
+
+  displayPostContent(content) {
+    // Clear existing content
+    this.innerHTML = "";
+
+    const postUrl = this.activityData.object;
+    // Create and append the distributed-post component
+    const distributedPostElement = document.createElement("distributed-post");
+    distributedPostElement.setAttribute("url", postUrl);
+    this.appendChild(distributedPostElement);
+  }
+
+  renderActivity() {
+    // Clear existing content
+    this.innerHTML = "";
+
+    switch (this.activityType) {
+      case "Create":
+        this.fetchAndDisplayPost(this.activityData.object);
+        break;
+      case "Follow":
+        this.displayFollowActivity();
+        break;
+      case "Like":
+        this.displayLikeActivity();
+        break;
+      default:
+        const message = `Activity type ${this.activityType} is not implemented yet.`;
+        const messageElement = document.createElement("p");
+        messageElement.textContent = message;
+        this.appendChild(messageElement);
+        break;
+    }
+  }
+
+  displayFollowActivity() {
+    const message = `New follow request from ${this.activityData.actor}`;
+    const messageElement = document.createElement("p");
+    messageElement.textContent = message;
+    this.appendChild(messageElement);
+  }
+
+  displayLikeActivity() {
+    const message = `New like on ${this.activityData.object}`;
+    const messageElement = document.createElement("p");
+    messageElement.textContent = message;
+    this.appendChild(messageElement);
+  }
+
+  attributeChangedCallback(name, oldValue, newValue) {
+    if ((name === "type" || name === "data") && newValue !== oldValue) {
+      this.activityType = this.getAttribute("type");
+      this.activityData = JSON.parse(this.getAttribute("data"));
+      this.renderActivity();
+    }
+  }
+}
+
+// Register the new element with the browser
+customElements.define("distributed-activity", DistributedActivity);
+
+// Functions to navigate to the next or previous page
+document.addEventListener("DOMContentLoaded", () => {
```

Comment:
Add these functions as methods on the outbox and and attach the button listeners in a separate script.